README.md

```diff
@@ -228,6 +228,7 @@ Options:
   --var-file FILENAME    A YAML file of variables to template into config
                          files.
   --ignore-dependencies  Ignore dependencies when executing command.
+  --merge-keys           Merge keys in successive var files not overwrite.
```

Comment:
Could you please add this to the documentation (in sceptre/docs folder) and provide a few examples how to do this and when you might want to do this?  i don't see of any of that info in issue #919 either.

Reply:
I just updated the docs to provide an example. There are more examples in the tests. Is that sufficient?

Reply:
tests are not docs nor are they examples.  users wanting to use this feature will most likely never try to understand your tests.   Also your example does not explain the purpose of this new flag.  It was not at all obvious to me why and when I would want to use this flag.

Also the description containing `..not overwrite` isn't really true, correct?  Your tests show that vars do get overrriden when using this flag. 

Another issue that may require further discussion is possibly making this functionality the default instead of requiring the user to specify `--merge-keys` flag.  I'm not sure I see any issue with that but others may disagree.  I would suggest you get on [#sceptre slack channel](https://sceptre.cloudreach.com/2.6.2/index.html?highlight=slack#communication) to drive this discussion.

Reply:
@zaro0508 I have updated the docs. I can't think of a better wording than 

```
--merge-keys           Merge keys in successive var files not overwrite.
```

The existing behaviour is for keys found in the last `var_file` to overwrite any that were defined in earlier ones. The new behaviour when `--merge-keys` is specified is to merge these keys instead of overwriting them. It is true that in some scenarios some subkeys still end up being overwritten according to the deep merge algorithm but I can't think of any easy way to express that in a few words. I am open to suggestions.